imagesloaded.js

```diff
@@ -198,37 +200,48 @@ ImagesLoaded.prototype.addBackground = function( url, elem ) {
 
 ImagesLoaded.prototype.check = function() {
   var _this = this;
-  this.progressedCount = 0;
+  this.completedCount = 0;
   this.hasAnyBroken = false;
+  this.intervalCheck = null;
+
   // complete if no images
   if ( !this.images.length ) {
     this.complete();
     return;
   }
 
-  function onProgress( image, elem, message ) {
-    // HACK - Chrome triggers event before object properties have changed. #83
-    setTimeout( function() {
-      _this.progress( image, elem, message );
+  this.runCheck = function() {
+
+    function onProgress(image, elem, message) {
+      // HACK - Chrome triggers event before object properties have changed. #83
+      setTimeout(function() {
+        _this.progress(image, elem, message);
+      });
+    }
+
+    _this.images.forEach(function(loadingImage) {
+      if (!loadingImage.isLoaded) {
+        loadingImage.once('progress', onProgress);
+        loadingImage.check();
+      }
     });
-  }
+  };
 
-  this.images.forEach( function( loadingImage ) {
-    loadingImage.once( 'progress', onProgress );
-    loadingImage.check();
-  });
+  this.intervalCheck = setInterval(this.runCheck, 500);
```

Comment:
Sorry for the delay in getting back to you.
I agree - I'd prefer things to be event based wherever possible. I've forgotten the logic behind using a timer now; I'll see if I can find some time to look at it some more.